Stripe/STPRedirectContext.m

```diff
@@ -154,14 +157,37 @@ - (void)safariViewControllerDidFinish:(__unused SFSafariViewController *)control
 }
 
 - (void)safariViewController:(__unused SFSafariViewController *)controller didCompleteInitialLoad:(BOOL)didLoadSuccessfully {
+    /*
+     SafariVC is, imo, over-eager to report errors. The way that (for example) girogate.de redirects
+     can cause SafariVC to report that the initial load failed, even though it completes successfully.
+
+     So, only report failures to complete the initial load if the host was a Stripe domain.
+     Stripe uses 302 redirects, and this should catch local connection problems as well as
+     server-side failures from Stripe.
+
+     We can only track the latest URL loaded on iOS 11, because `safariViewController:initialLoadDidRedirectToURL:`
+     didn't exist prior to that.
+     */
     if (didLoadSuccessfully == NO) {
-        stpDispatchToMainThreadIfNecessary(^{
-            [self handleRedirectCompletionWithError:[NSError stp_genericConnectionError]
-                        shouldDismissViewController:YES];
-        });
+        if (@available(iOS 11, *)) {
+            stpDispatchToMainThreadIfNecessary(^{
```

Comment:
We don't do anything here on iOS < 11?

Reply:
If there was some way to distinguish between real failures and spurious ones, I'd love to. I wasn't able to find one though.

Reply:
I had a similar initial reaction mostly by the way it's written. Maybe it needs an else clause with a doc:

```
stpDispatchToMainThreadIfNecessary(^{
  if (didLoadSuccessfully == NO) {
    if (@available(iOS 11, *) && [self.latestSafariVCUrl.host containsString:@"stripe.com"]) {
    // Report load errors that exist on `stripe.com` website
    ...
    } else {
      // Do nothing because `latestSafariVCUrl` is not accurate `< iOS 11`
    }
  }
}
```

In any case, this looks functional and don't have strong opinion on this one.